src/middlewared/middlewared/plugins/mdns.py

```diff
@@ -15,6 +17,74 @@
 from middlewared.service import Service
 
 
+class mDNSDaemonThread(object):
+    def __init__(self, *args, **kwargs):
+        super(mDNSDaemonThread, self).__init__()
+        self.mdnsd_pidfile = "/var/run/mdnsd.pid"
+        self.mdnsd_piddir = "/var/run/"
+
+    def pidfile_exists(self):
+        return os.access(self.mdnsd_pidfile, os.F_OK)
+
+    def is_alive(self):
+        mdnsd_proc = None
+        alive = False
+
+        if not self.pidfile_exists():
+            return False
+
+        for proc in psutil.process_iter():
+            procdict = proc.as_dict(['name'])
+            procname = str(procdict['name'])
+
+            match = re.search("([^/]+)$", procname)
+            if not match:
+                continue
+
+            procname = match.group(0)
+            if procname == "mdnsd":
+                mdnsd_proc = proc
+                break
+
+        if mdnsd_proc and mdnsd_proc.is_running():
+            alive = True
+
+        return alive
+
+    def wait(self):
```

Comment:
`wait` for what? Maybe it should be `wait_startup`?

Reply:
Wait for mDNSResponder to startup